kotlin-native/backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/objcexport/ObjCExportCodeGenerator.kt

```diff
@@ -1785,8 +1785,8 @@ private fun ObjCExportCodeGenerator.createDirectAdapters(
         )
     }
 
-    val inheritedAdapters = superClass?.getAllRequiredDirectAdapters().orEmpty()
-    val requiredAdapters = typeDeclaration.getAllRequiredDirectAdapters() - inheritedAdapters
+    val inheritedAdapters = superClass?.getAllRequiredDirectAdapters().orEmpty().toSet()
+    val requiredAdapters = typeDeclaration.getAllRequiredDirectAdapters().filter { !inheritedAdapters.contains(it) }
```

Comment:
So I need to convert to set to make the contains call fast, otherwise it's essentially calling contains on a list in a loop. 

If your question is more about why did I use filter instead of just calling minus, I'm not sure. I could just move back to calling minus on the set instead of a hand written filter.

Reply:
> If your question is more about why did I use filter instead of just calling minus, I'm not sure. I could just move back to calling minus on the set instead of a hand written filter.

Yes, it was exactly about this. I was not sure if you did this on purpose and what purpose.